include/bout/solver.hxx

```diff
@@ -324,6 +324,13 @@ public:
   /// The run from which this was restarted. Throws if the identifier hasn't been set yet.
   std::string getRunRestartFrom() const;
 
+  /// Get the number of completed output steps
+  int getIterationCounter() const { return iteration; }
+
+  /// Add one to the iteration count, used by BoutMonitor, but could be called by a
+  //user-defined monitor (if `bout_run()` is not used)
+  void incrementIterationCounter() { ++iteration; }
```

Comment:
Small suggestion:

```suggestion
  /// Add one to the iteration count, used by BoutMonitor, but could be called by a
  /// user-defined monitor (if `bout_run()` is not used).
  /// Returns the incremented value
  int incrementIterationCounter() { return ++iteration; }
```

means below we can do:

```cpp
  // Get updated iteration count
  iteration = solver->incrementIterationCounter();
```

Reply:
Nice! I'll put that in.